services/collateral.go

```diff
@@ -28,25 +29,39 @@ func (s *Service) GetCollateralPrice(blockNumber *big.Int, collateralType common
 }
 
 func (s *Service) RetrieveCollateralWithdrawnLimit(limit uint64) ([]*models.CollateralWithdrawn, error) {
-	iterations, last, err := s.getIterationsForLimitQueryCore(limit)
+	return s.RetrieveCollateralWithdrawn(0, 0, limit)
+}
+
+func (s *Service) RetrieveCollateralWithdrawn(fromBlock uint64, toBlock uint64, limit uint64) ([]*models.CollateralWithdrawn, error) {
+	iterations, lastBlock, err := s.getIterationsForLimitQueryCore(limit)
```

Comment:
should be `s.getIterationsForQuery`, instead of `s.getIterationsForLimitQueryCore`

Because `s.getIterationsForLimitQueryCore` gives iterations in range from core contract's first block to RPC's last block